feat: Redo getting started guide (#265)
serverpod/serverpod_docs: docs/01-get-started/04-working-with-the-database.md

```bash
$ cd magic_recipe/magic_recipe_server
$ serverpod create-migration
```

You will notice that there will be a new entry in your "migrations" folder - serverpod creates these migrations "step by step" - each time you have changes which are relevant to the database and run `serverpod create-migrations` a new migration file will be created. This is a good way to keep track of the changes you make to the database and to be able to roll back changes if needed.

## Writing to the database

Let's save all created recipes to the database. First we want to automatically store new recipes in the database. Because the `Recipe` now has a `table` it is now not just a serializable entity, but also a `TableRow` - that means that we can use the `insertRow` method to store it in the database.

```dart
// recipe_endpoint.dart

class RecipeEndpoint extends Endpoint {
  Future<Recipe> generateRecipe(Session session, String ingredients) async {

  // ...

    final recipe = Recipe(
      author: 'Gemini',
      text: responseText,
      date: DateTime.now(),
      ingredients: ingredients,
    );

    // --- Add this ---

    // Save the recipe to the database, but don't block the response

>     unawaited(Recipe.db.insertRow(session, recipe));

Author: I see, then we should probably also return the object from the database which includes the id

    // --- End of added code ---

    return recipe;
  }

```

## Reading from the database

Next we want to add a new endpoint that will return all the favourite recipes from the database. We will create a new method in the `RecipeEndpoint` class that will return all the recipes from the database.

To make sure that we get them in the correct order, we can sort them by the date they were created.

```dart
// recipe_endpoint.dart

class RecipeEndpoint extends Endpoint {

  // ...

  /// This method is used to get all the generated recipes from the database.
  Future<List<Recipe>> getRecipes(Session session) async {

>     // Get all the recipes from the database, sorted by date

Reviewer: ```suggestion
    // Get all the recipes from the database, sorted by date.
```

    return Recipe.db.find(
      session,
      orderBy: (t) => t.date,
      orderDescending: true,
    );
  }
```

:::info
The when adding a `table` to the model class definition, the model will now give you access to the database, specifically to the `recipes` table through `Recipe.db` (e.g. `Recipe.db.find(session)`.
The `insertRow` method is used to insert a new row in the database. The `find` method is used to query the database and get all the rows of a specific type. See [CRUD](../concepts/database/crud) and [relation queries](../concepts/database/relation-queries) for more information.
:::

## Generate the code

> Like before, when you change something that has an effect on the client code, you need to run `serverpod generate` - we don't need to run `serverpod create-migrations` again because we already created the migrations in the previous step.

Reviewer: ```suggestion
Like before, when you change something that has an effect on the client code, you need to run `serverpod generate`. We don't need to run `serverpod create-migrations` again because we already created the migrations in the previous step.
```

                    errorMessage: _errorMessage,
                  ),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}

/// _ResultDisplays shows the result of the call. Either the returned result from
/// the `example.hello` endpoint method or an error message.
class ResultDisplay extends StatelessWidget {
  final String? resultMessage;
  final String? errorMessage;

  const ResultDisplay({
    super.key,
    this.resultMessage,
    this.errorMessage,
  });

  @override
  Widget build(BuildContext context) {
    String text;
    Color backgroundColor;
    if (errorMessage != null) {
      backgroundColor = Colors.red[300]!;
      text = errorMessage!;
    } else if (resultMessage != null) {
      backgroundColor = Colors.green[300]!;
      text = resultMessage!;
    } else {
      backgroundColor = Colors.grey[300]!;
      text = 'No server response yet.';
    }

    return ConstrainedBox(
      constraints: const BoxConstraints(minHeight: 50),
      child: Container(
        color: backgroundColor,
        child: Center(
          child: Text(text),
        ),
      ),
    );
  }
}
```

## Run the app

> First we need to start the server AND apply the migrations:

Reviewer: ```suggestion
First, we need to start the server and apply the migrations by adding the `--apply-migrations` flag:
```

```bash
$ cd magic_recipe/magic_recipe_server
$ docker-compose up -d
$ dart run bin/main.dart --apply-migrations
```

> 

Reviewer: ```suggestion

:::tip
When developing your server, it's always safe to pass the `--apply-migrations` flag. If no migration needs to be applied, the flag is simply ignored. In a production environment, you may want to have a bit more control.
:::
```

> Then we can start the Flutter app:

Reviewer: ```suggestion
Now, start the Flutter app:
```

```bash
$ cd magic_recipe/magic_recipe_flutter
$ flutter run -d chrome
```

## Summary

> You have now learned about the fundamentals of Serverpod - how to create and use endpoints with custom data models and how to store the data in a database. You have also learned how to use the generated client code to call the endpoints from your Flutter app. You can now use this knowledge to create your own endpoints and data models and store them in a database.

Reviewer: ```suggestion
You now know the fundamentals of Serverpod - how to create and use endpoints with custom data models and how to store the data in a database. You have also learned how to use the generated client code to call the endpoints from your Flutter app. We cannot wait to see what you will build with Flutter and Serverpod.
```

> If you get stuck, never be afraid to ask questions in our [community on Github](https://github.com/serverpod/serverpod/discussions). The Serverpod team is very active there, and many questions are also answered by other developers in the community.

Reviewer: ```suggestion
If you get stuck, never be afraid to ask questions in our [community on Github](https://github.com/serverpod/serverpod/discussions). The Serverpod team is very active there, and many questions are also answered by other developers.
```

:::tip

> Working with a database is an extensive subject. Learn more in the [Database](../concepts/database/connection) section.

Reviewer: ```suggestion
Working with a database is an extensive subject, and Serverpod's ORM is very powerful. Learn more in the [Database](../concepts/database/connection) section.
```
